src/connector.rs

```diff
@@ -16,10 +16,13 @@ pub struct Connector {
 
 impl Connector {
     pub fn new() -> Self {
-        let config = Config::init();
+        let mut config = Config::init();
         let manifest = Manifest::init();
         let id = Connector::id(&manifest, &config);
         let producer = Producer::new(&config.kafka_url, &id);
+        let sub_config = &config.value[id];
```

Comment:
Create a method like `build_sub_config(&self, id &str)` in the `Config` struct to build sub_config